pkg/telemetry/setup.go

```diff
@@ -0,0 +1,96 @@
+package telemetry
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/jaeger"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	tracesdk "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+)
+
+// Global internal tracer that generates spans.
+var tracer = otel.Tracer("undefined")
```

Comment:
Initially, the idea was that I stored the tracer in a private global variable so that those who want to use `NewTelemetry()` don't need to supply an additional parameter called "tracer" (that they would need to either create or get it from the parent). But then I figured that `otel` actually sets a global tracer whenever we initialize and explicitly set it in the set-up function, i.e. we could access this tracer any time by just calling `otel.GetTracingProvider().Tracer()`, so there is actually no need to have an additional global variable in our case (also global variables are bad 🙂), so I removed this part completely! Thanks for pointing it out!